native/src/routes/EventsContainer.tsx

```diff
@@ -1,232 +1,107 @@
-import { reportError } from '../utils/sentry'
+import navigateToLanguageChange from '../navigation/navigateToLanguageChange'
 import Events from './Events'
+import LoadingErrorHandler from './LoadingErrorHandler'
 
-type EventsNavigationProps = {
+type NewEventsContainerProps = {
   route: RouteProps<EventsRouteType>
   navigation: NavigationProps<EventsRouteType>
 }
-type OwnProps = EventsNavigationProps
-type DispatchProps = {
-  dispatch: Dispatch<StoreActionType>
-}
-type EventsContainerProps = EventsNavigationProps &
-  DispatchProps & {
-    path: string | null | undefined
-    events: Array<EventModel>
-    cityModel: CityModel
-    language: string
-    resourceCache: LanguageResourceCacheStateType
-    resourceCacheUrl: string
-    refresh: (dispatch: Dispatch<StoreActionType>) => () => void
-  }
-type RefreshProps = EventsNavigationProps & {
-  cityCode: string
-  language: string
-  path: string | null | undefined
-}
-type StateProps = StatusProps<EventsContainerProps, RefreshProps>
 
-const refresh = (refreshProps: RefreshProps, dispatch: Dispatch<StoreActionType>) => {
-  const { route, navigation, cityCode, language, path } = refreshProps
-  const navigateTo = createNavigate(dispatch, navigation)
-  navigateTo(
-    {
+const EventsContainer = ({ navigation, route }: NewEventsContainerProps): ReactElement => {
+  const { slug } = route.params
+  const { cityCode, languageCode } = useCityAppContext()
+  const dispatch = useDispatch()
+  const resourceCacheUrl = useContext(StaticServerContext)
+
+  const response = useLoadEvents({ cityCode, languageCode })
+  const { data, refresh } = response
+
+  const currentEvent = slug ? data?.events.find(it => it.slug === slug) : undefined
+
+  useEffect(() => {
+    const goToLanguageChange = data
+      ? () => {
+          navigateToLanguageChange({
+            navigation,
+            languageCode,
+            languages: data.languages,
+            cityCode,
+            availableLanguages: currentEvent
+              ? Object.keys(currentEvent.availableLanguageSlugs)
+              : data.languages.map(it => it.code),
+          })
+        }
+      : undefined
+    navigation.setOptions({
+      // Only run on use effect dependency changes which means it is re-rendered anyway since props change
+      // eslint-disable-next-line react/no-unstable-nested-components
+      header: () => (
+        <Header
+          route={route}
+          navigation={navigation}
+          peeking={false}
+          categoriesAvailable
+          language={languageCode}
+          routeCityModel={data?.city}
+          goToLanguageChange={goToLanguageChange}
+        />
+      ),
+    })
+  }, [route, navigation, cityCode, languageCode, data, currentEvent])
+
+  const onLanguageChange = useCallback(
+    (newLanguage: string) => {
+      if (currentEvent) {
+        const newSlug = currentEvent.availableLanguageSlugs[newLanguage]
+        // TODO IGAPP-636: Handle language not available?
```

Comment:
I think it may already also be fine to just show the event list again if there is no translation in the new language?

Reply:
@sarahsporck @f1sh1918 @SteffiStuffel opinions on this? What should we do if a site that is opened in the background is not available in the new language? Just going back to the list of events/home page for categories is good enough I think. Other opinions?